### ISSUE TITLE
Add support for controlling a physical turtle

### ISSUE THREAD

**Issue by RoganDawes** [inexorabletash/jslogo#121]
Hi,

I'm looking to build a simple 2 wheel robot, controlled by an ESP8266 WiFi microcontroller. My idea is to host jslogo on the ESP8266's webserver, and provide a websocket endpoint where the interpreted instructions can be sent for execution by the robot. The idea would then be that the user would see the turtle perform each step in their browser, followed by the robot performing it. Once the robot has finished the step, the browser would continue to the next step.

I'm not much of a JavaScript programmer, could anyone give me an idea of how easy or difficult it might be to extend this code to incorporate the above functionality? Would it require a complete rewrite to allow that sort of interaction with the server, or is it a relatively simple project?

**Comment by inexorabletash** [comment]
It's probably easiest to consider running Logo on the controlling page itself and the physical Turtle as just the turtle. It would hook into the appropriate commands issued to the canvas turtle implementation.

**Comment by RoganDawes** [comment]
Yeah, I was thinking of running the two turtles in parallel, perhaps with a
“Test” button to run it only in the browser, and an “Execute” button to
send it to the robot as well. I guess sending it to the robot INSTEAD of
the canvas would work as well, on the Execute button.

Thanks for your response!
